keras_rs/src/layers/retrieval/remove_accidental_hits.py

```diff
@@ -0,0 +1,83 @@
+import keras
+import numpy as np
+from keras import ops
+
+from keras_rs.src import types
+from keras_rs.src.api_export import keras_rs_export
+from keras_rs.src.utils import keras_utils
+
+SMALLEST_FLOAT = np.finfo(np.float32).smallest_normal / 100.0
+
+
+@keras_rs_export("keras_rs.layers.RemoveAccidentalHits")
+class RemoveAccidentalHits(keras.layers.Layer):
+    """Zeroes the logits of accidental negatives.
+
+    Zeroes the logits of negative candidates that have the same ID as the
+    positive candidate in that row.
+    """
+
+    def call(
+        self,
+        labels: types.Tensor,
+        logits: types.Tensor,
+        candidate_ids: types.Tensor,
+    ) -> types.Tensor:
+        """Zeroes selected logits.
+
+        For each row in the batch, zeroes the logits of negative candidates that
+        have the same ID as the positive candidate in that row.
+
+        Args:
+          labels: one-hot labels tensor, typically
+              `[batch_size, num_candidates]` but can have more dimensions or be
+              1d as `[num_candidates]`.
+          logits: logits tensor. Must have the same shape as `labels`.
+          candidate_ids: candidate identifiers tensor, can be `[num_candidates]`
+              or `[batch_size, num_candidates]` or have more dimensions as long
+              as they match the last dimensions of `labels`.
```

Comment:
Hmmm, the other layers (`SamplingProbability`, `HardNegativeMining`) don't have the option to have extra batch dimensions. Should we do this for uniformity?

Reply:
Hmm... yeah, since they're used together.

I noticed it was easy to generalize so I just did it (although it made unit tests trickier).

I'll do it in a separate PR. https://github.com/keras-team/keras-rs/issues/39